File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Endpoint.py
from typing import Optional, List
from base import Period
from base import CodeableConcept
from base import Coding
from base import ContactPoint
from base import DomainResource
from base import Reference
from base import Identifier

class Endpoint(DomainResource):
    connection_type: Coding
    address: str
    managing_organization: Optional[Reference] = None
    name: Optional[str] = None
    payload_mime_type: Optional[List[str]] = None
    payload_type: list[CodeableConcept] = []
    header: Optional[List[str]] = None
    status: str
    identifier: Optional[List[Identifier]] = None
    period: Optional[Period] = None
    contact: Optional[List[ContactPoint]] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/Endpoint.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Endpoint : DomainResource, IResource
{
    public required Base.Coding ConnectionType { get; set; }
    public required string Address { get; set; }
    public Base.ResourceReference? ManagingOrganization { get; set; }
    public string? Name { get; set; }
    public string[]? PayloadMimeType { get; set; }
    public required Base.CodeableConcept[] PayloadType { get; set; }
    public string[]? Header { get; set; }
    public required string Status { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.Period? Period { get; set; }
    public Base.ContactPoint[]? Contact { get; set; }
}